tests/transformations/transient_reuse_test.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+from __future__ import print_function
+
+import argparse
+import dace
+import numpy as np
+from dace.transformation.interstate.transient_reuse import TransientReuse
+
+M = dace.symbol('M')
+N = dace.symbol('N')
+
+@dace.program
+def operation(A: dace.float64[M, M], B: dace.float64[M, M], C: dace.float64[M, N], D: dace.float64[M, N]):
+    tmp = dace.define_local([M, M, M], dtype=A.dtype)
+    E = dace.define_local([M,M], dtype=A.dtype)
+
+    @dace.map(_[0:M, 0:M, 0:M])
+    def multiplication(i, j, k):
+        in_A << A[i, k]
+        in_B << B[k, j]
+        out >> tmp[i, j, k]
+
+        out = in_A * in_B
+    dace.reduce(lambda a, b: a + b, tmp, E, axis=2, identity=0)
+
+    C[:] = A @ E @ (A @ B) @ (B @ D)
+
+
+if __name__ == "__main__":
+    print("==== Program start ====")
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("M", type=int, nargs="?", default=5)
+    parser.add_argument("N", type=int, nargs="?", default=100)
+    args = vars(parser.parse_args())
+
+    M.value = args["M"]
+    N.value = args["N"]
+
+    # Initialize arrays: Randomize A and B, zero C
+    A = np.random.rand(M.value, M.value).astype(np.float64)
+    B = np.random.rand(M.value, M.value).astype(np.float64)
+    D = np.random.rand(M.value, N.value).astype(np.float64)
+    C = np.zeros([M.value, N.value], dtype=np.float64)
+    C_regression = np.zeros_like(C)
+
+    sdfg = operation.to_sdfg(args)
```

Comment:
Can I somehow get M and N here into the operation? The memory savings output would look nicer...

Reply:
What does this mean? also what is `M/N.value`?

Reply:
I thought there may be a way to pass the values of M and N to operation.to_sdfg() so that when I sum the transient array sizes I actually get a value instead of combinations of M and N. Another option is to just set the sizes instead of having M and N as variables.
I should use M/N.value no?

Reply:
Should now be ok I think